internal/manifests/compactor_test.go

```diff
@@ -54,3 +54,13 @@ func TestNewCompactorStatefulSet_HasTemplateConfigHashAnnotation(t *testing.T) {
 	require.Contains(t, annotations, expected)
 	require.Equal(t, annotations[expected], "deadbeef")
 }
+
+func TestNewCompactorServiceMonitor_SelectorMatchesLabels(t *testing.T) {
+	sts := manifests.NewCompactorServiceMonitor("abcd", "efgh")
+
+	l := sts.GetObjectMeta().GetLabels()
+	for key, value := range sts.Spec.Selector.MatchLabels {
+		require.Contains(t, l, key)
+		require.Equal(t, l[key], value)
+	}
```

Comment:
This test is not addressing the core label matching issue with ServiceMonitors. The key thing to check here is that `ServiceMonitor.Spec.Selector.matchLabels` is matching the `Service.Spec.Selector`. This ensures that whenever we change the one we need to change the other to keep scraping working.